autoscale/src/main/java/com/sequenceiq/periscope/service/ScalingActivityService.java

```diff
@@ -40,6 +53,40 @@ public class ScalingActivityService {
     @Inject
     private RegionAwareCrnGenerator regionAwareCrnGenerator;
 
+    @Inject
+    private AutoscaleRestRequestThreadLocalService restRequestThreadLocalService;
+
+    @Inject
+    private ClusterService clusterService;
+
+    @Inject
+    private CloudbreakCommunicator cloudbreakCommunicator;
+
+    @Override
+    public AuthorizationResourceType getSupportedAuthorizationResourceType() {
+        return AuthorizationResourceType.DATAHUB;
+    }
+
+    @Override
+    @Retryable(value = NotFoundException.class, maxAttempts = 3, backoff = @Backoff(delay = 5000))
```

Comment:
Why is a backoff / retry defined for this one API only ? Trying to understand the scenario where this gets invoked without an entry existing in the database.

Reply:
We have defined retry for this as this would be called at the time of authorization. In the scenario where there are no entries present, it will be creating a cluster using the stack but as there are no scaling activities defined for this it will return 404 error at the end.